### DISCUSSION TITLE
ESP8266 MQTTClient poll check. or wait.msg

### DISCUSSION THREAD

**Discussion by Fisherman48** [micropython/micropython#12038] [ESP8266]
Hi
Using ESP8266-12 and 
**from umqtt.robust import MQTTClient**
From what I understand I have to explicit poll in a loop, for message, **using client.check_msg()** or **client.wait_msg()**
Are there any other implicit like background/ interrupt driven mechanism to activate the callnack: **client.set_callback(sub_cb)** ?

For ex for RasberryPye there are a mechanism for that:
**import paho.mqtt.client as mqtt #import the client**
**client.loop_start()**

Br /Stefan in the archipellago of Stockholm

**Comment by cwalther** [comment]
> Are there any other implicit like background/ interrupt driven mechanism to activate the callnack

No, not with `umqtt`.

You may be interested in a more sophisticated asynchronous MQTT client: https://github.com/peterhinch/micropython-mqtt/blob/master/mqtt_as/README.md

**Reply by Fisherman48** [reply]
Thanks,
 so, to stay on ESP8266 with limited memory I have to freeze the mqtt_as, or move to esp32.
Br /Stefan

**Comment by glenn20** [comment]
I haven't tried it personally, but you could try invoking `client.check_msg()` from a [Timer](https://docs.micropython.org/en/latest/esp8266/quickref.html#timers) callback if you are happy with the latency. I'm also not sure if there are restrictions on what code can be executed from a callback on esp8266.
